app/components/search.tsx

```diff
@@ -81,12 +81,16 @@ export default function Search({onSiteAnswersRef, openQuestionTitles, onSelect}:
   const model = tfFinishedLoadingRef.current ? 'tensorflow' : 'plaintext'
 
   const hideSearchResults = () => setShowResults(false)
+  const [hideEnabled, enableHide] = useState(true)
   const handleBlur = (e: FocusEvent<HTMLDivElement>) => {
     // If the focus changes from something in the search widget to something outside
     // of it, then hide the results. If it's just jumping around the results, then keep
     // them shown.
     const focusedOnResult = e.relatedTarget?.classList.contains('result-item') || false
-    setShowResults(focusedOnResult)
+    // Safari doesn't provide related target info in the blur event, so this will
+    // make sure that it won't hide the search results right away, as that stops
+    // the clicked element from fireing (it gets destroyed before the click handler fires)
+    if (hideEnabled) setShowResults(focusedOnResult)
```

Comment:
I very much dislike this "solution". But I don't see a better way for safari to know whether the clicked item should close the search results. Selecting items via the keyboard worked fine, though...